python312Packages.cpe: init at 1.2.1

Tochiaha opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

fabaff commented on pkgs/development/python-modules/cpe/default.nix:
```diff
@@ -0,0 +1,31 @@
+  src = fetchFromGitHub {
+    owner = "nilp0inter";
+    repo = "cpe";
+    rev = "v${version}";
```
```suggestion
    rev = "refs/tags/v${version}";
```

fabaff commented on pkgs/development/python-modules/cpe/default.nix:
```diff
@@ -0,0 +1,31 @@
+
+  build-system = [ setuptools ];
+
+  dependencies = [ sphinx pytest ];
```
`pytest` seems misplaced.

fabaff commented on pkgs/development/python-modules/cpe/default.nix:
```diff
@@ -0,0 +1,31 @@
+
+  pythonImportsCheck = [ "cpe" ];
+
+  # no test
```
The upstream repo contains [tests](https://github.com/nilp0inter/cpe/tree/develop/tests).

fabaff commented on pkgs/development/python-modules/cpe/default.nix:
```diff
@@ -0,0 +1,31 @@
+  meta = {
+    description = "Common platform enumeration for python";
+    homepage = "https://github.com/nilp0inter/cpe";
+    license = lib.licenses.gpl3;
```
See https://github.com/nilp0inter/cpe/blob/develop/setup.py#L30

```suggestion
    license = lib.licenses.lgpl3Only;
```

fabaff commented on pkgs/development/python-modules/cpe/default.nix:
```diff
@@ -0,0 +1,31 @@
+    homepage = "https://github.com/nilp0inter/cpe";
+    license = lib.licenses.gpl3;
+    maintainers = with lib.maintainers; [ tochiaha ];
+    platforms = lib.platforms.all;
```
This is the default.

```suggestion
```

fabaff reviewed (commented):
Please use `nixfmt`. A while back everything under `python-modules` was reformated.

Tochiaha committed:
python312Packages.cpe: init at 1.2.1

Update pkgs/development/python-modules/cpe/default.nix

Co-authored-by: Fabian Affolter <mail@fabian-affolter.ch>
Update pkgs/development/python-modules/cpe/default.nix

Co-authored-by: Fabian Affolter <mail@fabian-affolter.ch>
Update pkgs/development/python-modules/cpe/default.nix

Co-authored-by: Fabian Affolter <mail@fabian-affolter.ch>
python312Packages.cpe: init at 1.2.1
python312Packages.cpe: init at 1.2.1

fabaff committed:
python312Packages.cpe: enable tests

fabaff commented:
Result of `nixpkgs-review pr 324744` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.cpe</li>
    <li>python311Packages.cpe.dist</li>
    <li>python312Packages.cpe</li>
    <li>python312Packages.cpe.dist</li>
  </ul>
</details>

fabaff reviewed (approved):
Looks good to me.
